azurerm/resource_arm_dev_test_lab_schedule.go

```diff
@@ -0,0 +1,489 @@
+package azurerm
+
+import (
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/services/devtestlabs/mgmt/2016-05-15/dtl"
+	"github.com/Azure/azure-sdk-for-go/services/scheduler/mgmt/2016-03-01/scheduler"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmDevTestLabSchedules() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmDevTestLabSchedulesCreateUpdate,
+		Read:   resourceArmDevTestLabSchedulesRead,
+		Update: resourceArmDevTestLabSchedulesCreateUpdate,
+		Delete: resourceArmDevTestLabSchedulesDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"location": azure.SchemaLocation(),
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"dev_test_lab_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"status": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  dtl.EnableStatusDisabled,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(dtl.EnableStatusEnabled),
+					string(dtl.EnableStatusDisabled),
+				}, true),
+			},
+
+			"task_type": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"weekly_recurrence": {
+				Type:     schema.TypeSet,
```

Comment:
since there's only a single element available here can we make this a TypeList instead:

```suggestion
				Type:     schema.TypeList,
```

Reply:
Done.